PWMR, LSCR1 and DMACR registers only need to be initialized once at startup, 
not every time imxfb_set_par()/imxfb_activate_var() are called (for example 
when using SDL).
Setting these registers in imxfb_set_par()/imxfb_activate_var() has the side
effect (for example) of shutting down the backlight when opening an 8bits
SDL screen.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: linux-2.6.29/drivers/video/imxfb.c
===================================================================
--- linux-2.6.29.orig/drivers/video/imxfb.c	2009-05-26 19:35:11.000000000 +0200
+++ linux-2.6.29/drivers/video/imxfb.c	2009-05-26 19:37:20.000000000 +0200
@@ -456,6 +456,7 @@
 	pr_debug("var: yres=%d vslen=%d um=%d bm=%d\n",
 		var->yres, var->vsync_len,
 		var->upper_margin, var->lower_margin);
+	pr_debug("var: bpp=%d\n", var->bits_per_pixel);
 
 #if DEBUG_VAR
 	if (var->xres < 16        || var->xres > 1024)
@@ -497,9 +498,6 @@
 	writel(SIZE_XMAX(var->xres) | SIZE_YMAX(var->yres),
 			fbi->regs + LCDC_SIZE);
 	writel(fbi->pcr, fbi->regs + LCDC_PCR);
-	writel(fbi->pwmr, fbi->regs + LCDC_PWMR);
-	writel(fbi->lscr1, fbi->regs + LCDC_LSCR1);
-	writel(fbi->dmacr, fbi->regs + LCDC_DMACR);
 
 	return 0;
 }
@@ -686,6 +684,12 @@
 		goto failed_cmap;
 
 	imxfb_set_par(info);
+	/* JB: The following parameters are only needed once at startup and so
+		removed from from imxfb_set_par()/imxfb_activate_var() */
+	writel(fbi->pwmr, fbi->regs + LCDC_PWMR);
+	writel(fbi->lscr1, fbi->regs + LCDC_LSCR1);
+	writel(fbi->dmacr, fbi->regs + LCDC_DMACR);
+
 	ret = register_framebuffer(info);
 	if (ret < 0) {
 		dev_err(&pdev->dev, "failed to register framebuffer\n");
